### PR TITLE
Experiment/truthfulqa

### PR DESCRIPTION
**Pull request by DavidUdell**
A minimum-viable-set-of-scripts to collect activations, train a sparse autoencoder on them, and interpret the autoencoder's standard-basis dimensions!

@montemac I've thought a lot about gradually modularizing this, adding in unit tests, and refactoring for efficiency. I probably want to do that before working extensively with the yet larger models. But I _also_ want to avoid eternal project creep and push some working code to start.

### CODE REVIEW

File: sparse_coder/autoencoder_var.py
Comment (DavidUdell) [inline sparse_coder/autoencoder_var.py:1]: Sorry @wusche1, I must have accidentally made this commit from your git username on the shared VPS!

**Comment by DavidUdell** [comment]
@montemac I'm going to cowboy merge this branch.

I checked the tests when you do a fresh install from `main` and a fresh install from `experiment/truthfulqa`. The only changes I've made to the repo globally are a couple new `setup.py` items, and none of those impact the tests.

However, independently, it looks like a lot of our tests are failing because of a problem with `transformer_lens` and `tuned_lens` `torch` dependencies! If you update to the latest `torch`, `transformer_lens` doesn't behave properly. But if you keep the `torch` version we specify, `tuned_lens` doesn't have the structure expected and the tests crash immediately! I suspect the winning solution is to eventually factor out `transformer_lens` completely.

**Comment by montemac** [comment]
Thanks @DavidUdell!  Sorry it took me ages to get to this PR.  

I think my main comment is that, in an ideal world, data and results (i.e. the contents of the `sparse_coding/logs` folder) wouldn't live in the source repo, but would live in e.g.`wandb` or another database of some kind (if they in fact need to be stored).  Can we remove these from the repo, or do they represent important archival data/results?

Thanks again!

**Comment by DavidUdell** [comment]
I can cut that out at the next refactor! The data isn't super important, but I figured it was best scientific practice to be logging-by-default to _somwhere_.

Either I'll move logging over to `wandb`, or just drop persistent logging if `wandb` ends up being too slow.

**Pull request merged**